### PR TITLE
#47 Modified second screen UI

### PR DESCRIPTION
**Pull request by NikitaSukhanov**
- added classes for drawing nonogram
- connected second screen UI with backend

### CODE REVIEW

File: app/src/main/res/layout/activity_result.xml [hunk 1]
```diff
@@ -8,6 +8,27 @@
     android:background="@drawable/img_background"
     tools:context=".ui.result.ResultActivity">
 
+    <include
+        android:id="@+id/toolbar_start"
```
Comment (maik-nack) [inline app/src/main/res/layout/activity_result.xml]: Please, change id to `toolbar_result`

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/ResultActivity.java [hunk 2]
```diff
@@ -26,6 +29,13 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_result);
         findViewById(R.id.button_save_thumb).setOnClickListener(this);
         findViewById(R.id.button_save_nng).setOnClickListener(this);
+        setSupportActionBar((Toolbar) findViewById(R.id.toolbar_start));
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/ResultActivity.java]: Please, change id to `toolbar_result`

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawer.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    public static int getDeviceHeight(Context context) {
+        WindowManager wm = (WindowManager) context
+                .getSystemService(Context.WINDOW_SERVICE);
+        Display display = wm.getDefaultDisplay();
+        int height = display.getHeight();
+        return height;
+    }
+
+    private void threadShutdown() {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawer.java:110]: Please, write a comment for the function

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawer.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+    public static int getDeviceWidth(Context context) {
+        WindowManager wm = (WindowManager) context
+                .getSystemService(Context.WINDOW_SERVICE);
+        Display display = wm.getDefaultDisplay();
+        int width = display.getWidth();
+        return width;
+    }
+
+    public static int getDeviceHeight(Context context) {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawer.java:99]: Please, write a comment for the function

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawer.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                + 2 * _nngDrawThread.LINES_WIDTH;
+        params.height = cellSize * nonogramField.getRows()
+                + 2 * _nngDrawThread.LINES_WIDTH;
+        setLayoutParams(params);
+
+        _nngDrawThread.setNonogramField(nonogramField);
+        _nngDrawThread.setCellSize(cellSize);
+    }
+
+    public static int getDeviceWidth(Context context) {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawer.java:86]: Please, write a comment for the function

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawer.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                               int height) {
+        //resize
+    }
+
+    @Override
+    public void surfaceDestroyed(SurfaceHolder holder) {
+        threadShutdown();
+    }
+
+    public void setNonogramField(Field nonogramField) {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawer.java:59]: Please, write a comment for the function

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        _background = background;
+    }
+
+    private void setPainterParams() {
+        _painter.setStyle(Paint.Style.FILL);
+        _painter.setStrokeWidth(LINES_WIDTH);
+        _painter.setColor(Color.BLACK);
+    }
+
+    private void drawNng(Canvas canvas) {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java:118]: Please, write a comment for the function

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+    public void setCellSize(int size) {
+        _cellSize = size;
+    }
+
+    public void setBackground(Drawable background) {
+        _background = background;
+    }
+
+    private void setPainterParams() {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java:107]: Please, write a comment for the function

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+
+    public void setNonogramField(Field nonogramField) {
+        _nonogram = nonogramField;
+    }
+
+    public void setCellSize(int size) {
+        _cellSize = size;
+    }
+
+    public void setBackground(Drawable background) {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java:100]: Please, write a comment for the function

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+    public void shutdown() {
+        _isRunning = false;
+    }
+
+    public void setNonogramField(Field nonogramField) {
+        _nonogram = nonogramField;
+    }
+
+    public void setCellSize(int size) {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java:91]: Please, write a comment for the function

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+                _surfaceHolder.unlockCanvasAndPost(canvas);
+            }
+        }
+    }
+
+    public void shutdown() {
+        _isRunning = false;
+    }
+
+    public void setNonogramField(Field nonogramField) {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java:82]: Please, write a comment for the function

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+            canvas = _surfaceHolder.lockCanvas(null);
+            if (canvas != null) {
+                canvas.drawColor(Color.WHITE);
+                drawNng(canvas);
+                _surfaceHolder.unlockCanvasAndPost(canvas);
+            }
+        }
+    }
+
+    public void shutdown() {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java:73]: Please, write a comment for the function

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawer.java [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+        int cellSizeHor = (getDeviceWidth(getContext()) - 2 * _minMarginHor)
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawer.java]: Please check line wrapping in coding standard

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawer.java [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+import android.view.SurfaceHolder;
+import android.view.SurfaceView;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+
+import com.mithridat.nonoconverter.backend.Field;
+
+public class NonogramDrawer extends SurfaceView implements SurfaceHolder.Callback {
+
+    private NonogramDrawThread _nngDrawThread;
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawer.java:18]: Please, write comments for the variables

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+import android.graphics.Paint;
+import android.graphics.RectF;
+import android.graphics.drawable.Drawable;
+import android.view.SurfaceHolder;
+
+import com.mithridat.nonoconverter.backend.Field;
+
+public class NonogramDrawThread extends Thread {
+
+    public final int LINES_WIDTH = 3;
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java:17]: Please, write comments for the variables

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+
+        // horizontal lines
+        for (int i = 0; i <= rows; i++) {
+            canvas.drawLine(startX, startY + i * _cellSize, stopX,
+                    startY + i * _cellSize, _painter);
+        }
+
+        //vertical lines
+        for (int i = 0; i <= columns; i++) {
+            canvas.drawLine(startX + i * _cellSize, startY,
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java]: Please check line wrapping in coding standard

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+        int radius = cellGap;
+        _painter.setColor(Color.BLACK);
+
+        //background, if present
+        if (_background != null)
+            _background.draw(canvas);
+
+        // horizontal lines
+        for (int i = 0; i <= rows; i++) {
+            canvas.drawLine(startX, startY + i * _cellSize, stopX,
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramDrawThread.java]: Please check line wrapping in coding standard

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/ResultActivity.java [hunk 18]
```diff
@@ -26,6 +29,13 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_result);
         findViewById(R.id.button_save_thumb).setOnClickListener(this);
         findViewById(R.id.button_save_nng).setOnClickListener(this);
+        setSupportActionBar((Toolbar) findViewById(R.id.toolbar_result));
+        NonogramDrawer nonogramDrawer = findViewById(R.id.nonogram_drawer);
+        Field field = getIntent()
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/ResultActivity.java]: Please check line wrapping in coding standard

**Review by maik-nack — APPROVED** [review]
OK

**Review by MaximSmolskiy — APPROVED** [review]
Normal

**Pull request merged**